documentation/service/technical/searching-by-location.md

```diff
@@ -0,0 +1,175 @@
+
+    We ignore the location. A nationwide location filter is irrelevant when all our vacancies are restricted to England.
+
+2. **We have a polygon** stored for the given location.
+
+    We create a buffered expanded area based on the provided search location radius.
+
+3. **No polygon is found** for the given location.
+
+    We get the geographical coordinates for the given location.
+
+## Location Polygons
+
+The location polygons are geographical areas we store in our database.
+
+There is an [ImportPolygonDataJob](/app/jobs/import_polygon_data_job.rb) running weekly that imports polygons for:
+- Counties
+- Cities
+- Regions
+
+It also creates composite polygons combining the above.
+
+All these imports are obtained [querying](/app/services/ons_data_import/base.rb) the ONS (Office for National Statistics) ArcGIS endpoints to obtain the area data for each of the polygons.
+
+### Purpose
+
+There is a difference between searching for vacancies within a particular distance from a jobseeker's home (that would match some particular coordinates point) and vacancies within a particular distance "from Essex".
+
+How do we measure the distance between a whole region and a particular vacancy location?
+
+Taking the center of the region would be wrong. As if, let's say, we search for "10 miles from Essex" anything over 10 miles from Essex center point would be filtered out.
+
+What we would expect is to find anything 10 miles away from Essex outer borders.
+
+Having an area stored for "Essex" containing its borders, allows us to combine `ST_Buffer` and `ST_DWithin` to first expand the area borders to cover the provided search radius, and then find if any vacancy location is contained within that expanded area.
+
+### Mappings
+
+Within the [mapping files](/config/data/ons_mappings/) we define the subset of cities, regions and counties we store polygon areas for. This is used by the [location data setup](/config/initializers/location_data.rb).
+
+The [mapped locations file](/config/data/ons_mappings/mapped_locations.yml) helps to match common location search terms with an appropriate location polygon if any.
+
+## Getting the geographical coordinates
+
+As many location search terms don't match a polygon name, the service falls back to obtaining the location coordinates.
+
+We use the [Geocoding class](/lib/geocoding.rb) to retrieve and cache the location coordinates. The default source for retrieving the location coordinates is the Google Geocoding API.
+
+Information about the API usage/costs should be accessible from the Google Cloud panel using your DfE email account.
+
+### Google Geocoding API costs
+
+Google Geocoding API is one of our higher costs in Google Cloud.
+To reduce the cost as much as possible, and as the coordinate info in the UK for a given point is quite stable data unlikely to change, we cache its responses for a long period, so there are fewer API hits renewing existing cached results.
```

Comment:
Any reason we don't say exactly how long it is cached for?

Reply:
I was concerned changes in the constant value would not be updated in the doc info and they could fall out of sync. But I have no problem adding it if provides better context about the magnitude of the cache length.